### DISCUSSION TITLE
Piped Downloads

### DISCUSSION THREAD

**Discussion by matt873** [deniscerri/ytdlnis#286] [Q&A]
Using Piped url from Libre tube. Will these downloads completely go through a piped proxy or just initial info data is fetched for the download card?
Also, the option to select Piped proxy for format sources in the app, that only fetches info data for download card?

**Comment by deniscerri** [comment]
Yes piped is only used for info data. The downloading is handled by ytdlp

**Comment by matt873** [comment]
Ok, I thought that would be the case when we select piped proxy from app settings but how will the app handle direct URL (https://piped.video/watch?...) pasted into the "search URL" box in the app home screen? Wouldn't yt-dlp try to do a download from the piped URL instead?

**Comment by deniscerri** [comment]
Hm yes you are correct. It will use that url. I should make the app use the YouTube url instead

**Comment by matt873** [comment]
No need for that. People like me who are using libretube to avoid connection through their real IP to YouTube servers will need it and removing it will take a useful function away from some users who want to achieve that level of anonymity and privacy. I think it should be the same as it is. Someone watching videos through a piped instance will also want to download videos through a piped instance and will not need direct download from YouTube servers and as an option, librtube has the option to use a direct YouTube URL for download.

**Reply by deniscerri** [reply]
Alright
